src/Monitor/Tracing/Zipkin.hs

```diff
@@ -361,11 +369,11 @@ serverSpan = serverSpanWith id
 -- | Generates a server span, optionally modifying the span's builder. This can be useful in
 -- combination with 'addEndpoint' if the remote client does not have tracing enabled.
 serverSpanWith :: MonadTrace m => (Builder -> Builder) -> B3 -> m a -> m a
-serverSpanWith f = incomingSpan "SERVER" (Endo f)
+serverSpanWith f b3 = incomingSpan "SERVER" (importB3 b3 <> Endo f)
 
 -- | Generates a child span with @CONSUMER@ kind. The producer's 'B3' should be provided as input.
```

Comment:
Do you mind expanding the doc to mention how IDs are handled? Something like "The generated span will have its parent ID set to the input B3's span ID.".

Reply:
Totally agree - but the text you suggested is related to `consumerSpanWith`. I will add a comment to both methods.

Reply:
FYI the comment was on line 374 (`consumerSpanWith`'s doc) - GitHub displays preceding lines as context.

Reply:
You see - I'm not really used to this process; please excuse my irritation 😢